src/xhr.js

```diff
@@ -63,6 +79,48 @@ class Xhr {
 
 
 /**
+ * Normalized method name by uppercasing.
+ * @param {string|undefined} method
+ * @return {string}
+ * @private
+ */
+export function normalizeMethod_(method) {
+  if (method === undefined) {
+    return 'GET';
+  }
+  return method.toUpperCase();
+}
+
+/**
+* Initialize init object with headers and stringifies the body.
+ * @param {!FetchInit} init
+ * @private
+ */
+function setupJson_(init) {
+  assert(allowedMethods_.indexOf(init.method) != -1, 'Only one of ' +
```

Comment:
In this scenario, this should go into our fetchPolyfill_. This code doesn't seem to care about validity. E.g.
```
if (init.method) {
  assert(...);
} else {
  init.method = 'GET';
}
```